### PR TITLE
[server] Improve GitHub Enterprise avatars handling

### PR DESCRIPTION
**Pull request by jankeromnes**
## Description

GitHub Enterprise doesn't seem to allow cross-origin requests for user avatars, leading to a broken Gitpod UI.

Also, I initially wanted to fetch the avatars, and encode them as data URIs, however GitHub Enterprise also doesn't seem to have an API to download avatars (only browser authentication via cookies seems to work).

Thankfully, GitHub Enterprise does provide a Gravatar ID that we can use. This Pull Request does just that whenever possible, and also improves the display of avatars that cannot be found or loaded.

## Related Issue(s)

Fixes https://github.com/gitpod-io/gitpod/issues/8755
Partially improves https://github.com/gitpod-io/gitpod/issues/8815

## How to test

1. Connect with GitHub Enterprise
2. Open the org selector in `/new`
3. The avatars should work for all accounts/orgs that use Gravatar
4. Add a project and start a prebuild for a repo where the owner does **not** use Gravatar
5. In the Project Branches & Prebuilds pages, the broken avatar will show the alt text, but it should be truncated such that the rest of the details are still legible

## Release Notes

```release-note
[server] Improve GitHub Enterprise avatars handling
```

### CODE REVIEW

**Comment by jankeromnes** [comment]
Hmm, of course, neither our GHE user nor its org have Gravatars, so that doesn't help 😅

Screenshot 2022-03-15 at 15 14 05
Screenshot 2022-03-15 at 15 13 47

But I guess this is the best we can do. I'll also fix the avatar width in the workspaces list and call it done. 😕

File: components/server/ee/src/prebuilds/github-service.ts [hunk 1]
```diff
@@ -32,8 +32,10 @@ export class GitHubService extends RepositoryService {
             return <ProviderRepository>{
                 name: r.name,
                 cloneUrl: r.clone_url,
-                account: r.owner?.login,
-                accountAvatarUrl: r.owner?.avatar_url,
+                account: r.owner.login,
+                accountAvatarUrl: r.owner.gravatar_id
+                    ? `https://www.gravatar.com/avatar/${r.owner.gravatar_id}?size=128`
```
Comment (gtsiolis) [inline components/server/ee/src/prebuilds/github-service.ts:37]: **issue**: This is still failing to load the avatar when there's no gravatar set, right?

| New project | Project branches |
|-|-|
| Screenshot 2022-03-15 at 8 00 55 PM | Screenshot 2022-03-15 at 8 00 43 PM |

**Review by gtsiolis — COMMENTED** [review]
I was in the neighborhood and left one comment below.
Thanks for tackling this, @jankeromnes!

File: components/server/ee/src/prebuilds/github-service.ts (hunk 1 above)
Comment (gtsiolis) [reply components/server/ee/src/prebuilds/github-service.ts:37]: Ah, I see.

**Review by jldec — APPROVED** [review]
merci @jankeromnes

**Pull request merged**